pytorch_lightning/trainer/data_loading.py

```diff
@@ -177,14 +177,19 @@ def reset_train_dataloader(self, model):
         self.is_iterable_train_dataloader = (
             EXIST_ITER_DATASET and isinstance(self.train_dataloader.dataset, IterableDataset)
         )
-        if self.is_iterable_train_dataloader and not isinstance(self.val_check_interval, int):
+        if self.is_iterable_dataloader(self.train_dataloader) and not isinstance(self.val_check_interval, int):
```

Comment:
this seems wrong. it needs to be self.is_iterable_train_dataloader as defined in 177

Reply:
or remove 177

Reply:
and what if thou train with another dataloader, checking if it is updated...
https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/trainer.py#L857

Reply:
i don't understand what you mean

Reply:
at what point is/was determined if the train dataset is iterable?

Reply:
ok, found it and it would be fine as it was... shall I rever ti?